### PR TITLE
Pico W grab bag

### PR DESCRIPTION
**Pull request by jepler**
* Fix large `send()` requests on stream packets. Now, they will correctly send only part of the buffer. Typically, 2920 bytes (2×MSS) can be sent on an idle socket, or 1460 bytes (1×MSS) on a socket that's recently seen a transmission. This has always been according to the documentation of `send` but previously send was expected to act like sendall by e.g., adafruit_httpserver. Closes: #7077 
 * Add `sendall()`, compatible with CPython, to send a whole buffer with multiple underlying `send` calls.
 * When lwip debugging is enabled, 16-bit values weren't formatted correctly, because "hu" was not a good string format for mp_printf
 * enabled resolution of `.local` MDNS names. This still doesn't enable picow to *HAVE* a ".local" MDNS name.
 * Implemented `wifi.Radio.stop_station`. After `stop_station`, the ip address is None. A new connection can be made with `connect`. There's something weird about the implementation, but I commented on it for future generations / me in 2 weeks from now.

### CODE REVIEW

File: shared-bindings/socketpool/Socket.c [hunk 1]
```diff
@@ -255,6 +255,39 @@ STATIC mp_obj_t _socketpool_socket_send(mp_obj_t self_in, mp_obj_t buf_in) {
...
+        // Bad file number.
+        mp_raise_OSError(MP_EBADF);
+    }
+    if (!common_hal_socketpool_socket_get_connected(self)) {
+        mp_raise_BrokenPipeError();
+    }
+    mp_buffer_info_t bufinfo;
+    mp_get_buffer_raise(buf_in, &bufinfo, MP_BUFFER_READ);
+    while (bufinfo.len > 0) {
+        mp_int_t ret = common_hal_socketpool_socket_send(self, bufinfo.buf, bufinfo.len);
```
Comment (dhalbert) [inline shared-bindings/socketpool/Socket.c:280]: Might this block and block background tasks? Should there be a `RUN_BACKGROUND_TASKS` here?

**Review by dhalbert — COMMENTED** [review]
One q

**Review by anecdata — COMMENTED** [review]
### wifi disconnection

After fresh reset, IPv4 properties populate immediately after connection, and de-populate immediately (to `None`) after a disconnect is triggered by `wifi.radio.stop_station()`. We should now be able to use the presence of an IPv4 address as a reliable indicator of the station being connected to an AP, consistent with`espressif` behavior. But note that connections are still currently maintained across reloads, unlike `espressif`.
```py
import time
import wifi
import socketpool
from secrets import secrets

pool = socketpool.SocketPool(wifi.radio)

def ipv4_info():
    return wifi.radio.ipv4_address, wifi.radio.ipv4_gateway, wifi.radio.ipv4_subnet, wifi.radio.ipv4_dns

print(f"{ipv4_info()}")
while True:
    print(f"Connecting... ", end="")
    wifi.radio.connect(secrets["ssid"], secrets["password"])
    print(f"{ipv4_info()}")
    print(f"Disconnecting... ", end="")
    wifi.radio.stop_station()
    print(f"{ipv4_info()}")
    time.sleep(1)
```
yields:
```
(None, None, None, None)  # if device was connected before a reload, it will show connected IPv4 info here
Connecting... (192.168.6.202, 192.168.4.1, 255.255.252.0, 1.1.1.2)
Disconnecting... (None, None, None, None)
# rinse, repeat
```

_Addendum:_ I've seen one case among dozens where the IPv4 info was not cleared after disconnect. I'll add some instrumentation to see how long it takes in that case.


### mDNS resolution

Using test code from [7032](https://github.com/adafruit/circuitpython/issues/7032#issue-1403897058), all four combinations of regular DNS / mDNS, and relative domain names / FQDNs, now resolve properly.

File: shared-bindings/socketpool/Socket.c (hunk 1 above)
Comment (jepler) [reply shared-bindings/socketpool/Socket.c:280]: Added

**Review by dhalbert — APPROVED** [review]
Thanks!

**Pull request merged**